sale_commission/models/account_invoice.py

```diff
@@ -117,6 +119,7 @@ class AccountInvoiceLineAgent(models.Model):
     settled = fields.Boolean(
         compute="_compute_settled",
         store=True, copy=False)
+    company_id = fields.Many2one('res.company', 'Company')
```

Comment:
No, you should take another approach: filter first the invoices that are for that company, and then select the lines.